src/main/java/ch/njol/skript/expressions/ExprRawString.java

```diff
@@ -0,0 +1,94 @@
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionList;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.VariableString;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+
+@Name("Raw String")
+@Description("Returns the string without formatting (colors etc.) and without stripping them from it, " +
+	"e.g. <code>raw \"&aHello There!\"</code> would output <code>&aHello There!</code>")
+@Examples("send raw \"&aThis text is unformatted!\" to all players")
+@Since("INSERT VERSION")
+public class ExprRawString extends PropertyExpression<String, String> {
+
+	static {
+		Skript.registerExpression(ExprRawString.class, String.class, ExpressionType.SIMPLE, "raw %strings%");
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<? extends String>[] messages;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr(LiteralUtils.defendExpression(exprs[0]));
+		messages = getExpr() instanceof ExpressionList<?> ?
+			((ExpressionList<String>) getExpr()).getExpressions() : new Expression[]{getExpr()};
+		for (Expression<? extends String> message : messages) {
+			if (message instanceof ExprColoured) {
+				Skript.error("the 'colored' expression may not be used with a 'raw string' expression");
+				return false;
+			}
+		}
+		return LiteralUtils.canInitSafely(getExpr());
+	}
+
+	@Override
+	protected String[] get(Event e, String[] source) {
+		List<String> strings = new ArrayList<>();
+		for (Expression<? extends String> message : messages) {
+			if (message instanceof VariableString) {
+				strings.add(((VariableString) message).toUnformattedString(e));
+				continue;
+			}
+			strings.addAll(Arrays.asList(message.getArray(e)));
+		}
+		return strings.toArray(new String[0]);
```

Comment:
Rather than building an empty array 0 which Java will have to re-initalize to it's size. You can just use `strings.toArray(String[]::new)` which will build an array based on the size. Java sees the 0, and says that ain't going to fit, so it builds a new array of the size. The way you did it is for when you know the size.

Reply:
Both are fine IMO, Skript has usages of both throughout its code

Reply:
> Rather than building an empty array 0 which Java will have to re-initalize to it's size. You can just use `strings.toArray(String[]::new)` which will build an array based on the size. Java sees the 0, and says that ain't going to fit, so it builds a new array of the size. The way you did it is for when you know the size.

For that I have to stream the array first `strings.stream.toArray(String[]::new)` and when I do so, Intellij just suggests `strings.toArray(new String[0])` as seen here

![image](https://user-images.githubusercontent.com/98935832/178518661-8bd91399-05bb-41ec-bc94-30f629a426da.png)

Reply:
The existing method should be significantly faster:
Dynamic array creation is a hotspot instrinsic candidate.
The String[]::new method reference requires a dynamic call to the lambda metafactory.

Reply:
The efficiency of both is negligible, so it really doesn't matter. At that point it's up to code conventions, which we don't have for this case. Either is fine, and both are used throughout Skript.